solutions/projectstaffing/jgraph/core/src/main/scala/com/microsoft/graphdataconnect/skillsfinder/service/UserService.scala

```diff
@@ -48,41 +49,59 @@ class UserService {
+  private val arrayAnonymousUserInfo = new Array[UserInfo](1)
+  arrayAnonymousUserInfo(0) = anonymousUserInfo
+
+
   private val logger: Logger = LoggerFactory.getLogger(classOf[UserService])
 
   def getUserId(cookieValue: String): String = {
     getUserInfo(cookieValue).userId
   }
 
   def getUserInfo(cookieValue: String): UserInfo = {
-    Try {
-      Unirest.get(jgraphUrl + "/.auth/me").headers(Map("Cookie" -> ("AppServiceAuthSession=" + cookieValue)).asJava).asJson()
-    } match {
-      case Success(response: HttpResponse[JsonNode]) =>
-        if (response.getStatus == HttpStatus.OK) {
-          val jsonResponse = response.getBody.toPrettyString
-
-          val arrayUserInfo: Array[UserInfo] = objectMapper.readValue(jsonResponse, classOf[Array[UserInfo]])
-          if (arrayUserInfo.nonEmpty) {
-            arrayUserInfo(0)
+    if (anonymousUser) {
+      arrayAnonymousUserInfo(0)
+    } else {
+      Try {
+        Unirest.get(jgraphUrl + "/.auth/me").headers(Map("Cookie" -> ("AppServiceAuthSession=" + cookieValue)).asJava).asJson()
+      } match {
+        case Success(response: HttpResponse[JsonNode]) =>
+          if (response.getStatus == HttpStatus.OK) {
+            val jsonResponse = response.getBody.toPrettyString
+
+            val arrayUserInfo: Array[UserInfo] = objectMapper.readValue(jsonResponse, classOf[Array[UserInfo]])
+            if (arrayUserInfo.nonEmpty) {
+              arrayUserInfo(0)
+            } else {
+              throw new UnauthorizedException("There was no information about the user that could be received from calling /.auth/me endpoint.")
+            }
           } else {
-            throw new UnauthorizedException("There was no information about the user that could be received from calling /.auth/me endpoint.")
+            throw new UnauthorizedException(s"Failed to get user info from /.auth/me endpoint. Status: ${response.getStatusText} Body: ${response.getBody}")
           }
-        } else {
-          throw new UnauthorizedException(s"Failed to get user info from /.auth/me endpoint. Status: ${response.getStatusText} Body: ${response.getBody}")
-        }
 
-      case Failure(e: Throwable) =>
-        logger.error("Failed to get user info from /.auth/me endpoint!", e)
-        throw new UnauthorizedException("Failed to get user info from /.auth/me endpoint!")
+        case Failure(e: Throwable) =>
+          logger.error("Failed to get user info from /.auth/me endpoint!", e)
+          throw new UnauthorizedException("Failed to get user info from /.auth/me endpoint!")
+      }
     }
+
   }
 
 
   def isUserPartOfAdminsGroup(clientPrincipalToken: String): Boolean = {
-    val jwtTokenHeaders = objectMapper.readValue(JwtTokenUtils.extractHeader(clientPrincipalToken), classOf[JwtTokenHeaders])
-    val userGroups = jwtTokenHeaders.claims.filter(_.typ.equals("groups"))
-    userGroups.exists(_.value.equals(gdcAdminsGroupId))
+    if (anonymousUser) {
+      true
```

Comment:
that true makes anonymous an admin user. It doesn't look correct

Reply:
This was solved in a past commit